### PR TITLE
fix(limit-count): different route with same conf should not share counter

### LINKED ISSUE #7746: bug: If the same configuration is used, different routes will share same limit for `limit-count`, even though they use different plugin_configs
### Current Behavior

as the title

### Expected Behavior

not share the limit 

### Error Logs

none

### Steps to Reproduce

1. Create two `plugin_config` with the same configuration

```
/apisix/plugin_configs/1
{"id":"1","plugins":{"limit-count":{"count":3,"key":"remote_addr","policy":"local","rejected_code":503,"time_window":60}}}
```
```
/apisix/plugin_configs/2
{"id":"2","plugins":{"limit-count":{"count":3,"key":"remote_addr","policy":"local","rejected_code":503,"time_window":60}}}
```

2. Create two routes, each using a different `plugin_config`

```
/apisix/routes/1
{"id":"1","plugin_config_id":"1","upstream_id":"1","uris":["/get"]}
```
```
/apisix/routes/2
{"id":"2","plugin_config_id":"2","upstream_id":"2","uris":["/anything"]}
```

4. Visit the route `/get` and you can see that there are 2 remaining left

```
curl 127.0.0.1:9080/get -i

...
X-RateLimit-Limit: 3
X-RateLimit-Remaining: 2
...
```

5. Visit the route `/anything`, this is only 1 remaining left (should have 2 left)
```
curl 127.0.0.1:9080/anything -i

...
X-RateLimit-Limit: 3
X-RateLimit-Remaining: 1
...
```

### Environment

- APISIX version (run `apisix version`): 2.13.2
- Operating system (run `uname -a`): Darwin
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): 1.19.3.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): 3.4.20
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Description

Fixes #7746

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/limit-count3.t [hunk 1]
```diff
@@ -224,3 +224,64 @@ passed
...
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/2',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "/hello1",
+                    "plugins": {
+                        "limit-count": {
+                            "count": 2,
+                            "time_window": 60
+                        }
```
Comment (tzssangglass) [inline t/plugin/limit-count3.t:243]: the conf of `limit-count` different form route 1

```
    "limit-count": {
        "count": 2,
        "time_window": 61
    }
```

is the same conf means the value of attr such as `count` or `time_window`?

**Comment by moonming** [comment]
is this PR will fix https://github.com/apache/apisix/issues/7746?

File: t/plugin/limit-count3.t (hunk 1 above)
Comment (spacewander) [reply t/plugin/limit-count3.t:243]: > is the same conf mean the value of attr such as count or time_window all equal to another limit-count in other routes?

Yes.

**Comment by spacewander** [comment]
> is this PR will fix #7746?

Yes

File: t/plugin/limit-count3.t (hunk 1 above)
Comment (tzssangglass) [reply t/plugin/limit-count3.t:243]: but the `time_window = 60` in TEST 7 is different form the `time_window = 61` TEST 5?

It seems that TEST 5&7 should test the same conf and not affect each other.

Comment (spacewander) [reply t/plugin/limit-count3.t]: Yes, you are right! Fixed.

**Review by tzssangglass — APPROVED** [review]

**Comment by nic-chen** [comment]
Hi @spacewander 
I have a doubt. Will this change affect `global_rule`? Will each route count its limit when configuring a `limit-count` in a `global_rule` ? Thanks.

**Comment by spacewander** [comment]
> Hi @spacewander I have a doubt. Will this change affect `global_rule`? Will each route count its limit when configuring a `limit-count` in a `global_rule` ? Thanks.

As all route shares the same global rule,  they will use the same limit counter in this global rule.

**Review by nic-chen — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**